### DISCUSSION TITLE
Timeframe for release with syn 2?

### DISCUSSION THREAD

**Discussion by hpenne** [asomers/mockall#494] [Questions]
Are there any plans for a new release that uses syn 2?  The master branch already seems to use it.  Other crates that I use have upgraded, so mockall is the last crate I use that still use syn 1.  It would be great to be able to use only one syn version.

**Comment by asomers** [comment]
No fixed time frame.  The only thing holding it up is the `#[concretize]` feature.  That still feels hacky, so I've resisted putting it into a release.  However, I haven't found any other way to solve that problem, so maybe I should go ahead?

**Reply by hpenne** [reply]
That's a difficult one to have an opinion on when I don't know the details of the implementation of this feature.  Generally, if it feels a little "off" right from the start, then I would probably recommend listening to your instincts and at least sleep on it for a little while.

**Comment by hpenne** [comment]
Is there any progress on this?  Would it be possible to release a version with syn 2 without the `#[concretize]` feature, if that is still not ready?

**Comment by asomers** [comment]
I think that concretize has been here long enough that it's not going to change much.  The only major change I would make at this point would require changing the `#[automock]` syntax to something like `#[Derive(automock)]`.  But that won't happen in the next release.  I'll go ahead and make a new release once #515 and #516 are done.

**Comment by asomers** [comment]
Mockall 0.12.0 is out now, using syn 2.
